fit.py

```diff
@@ -0,0 +1,78 @@
+from numpy import mean
+from numpy import std
+from sklearn.model_selection import KFold
+from keras.datasets import mnist
+from keras.utils import to_categorical
+
+import working_model
+
+
+# load train and test dataset
+def load_dataset():
+	# load dataset
+	(trainX, trainY), (testX, testY) = mnist.load_data()
+	# reshape dataset to have a single channel
+	trainX = trainX.reshape((trainX.shape[0], 28, 28, 1))
+	testX = testX.reshape((testX.shape[0], 28, 28, 1))
+	# one hot encode target values
+	trainY = to_categorical(trainY)
+	testY = to_categorical(testY)
+	return trainX, trainY, testX, testY
```

Comment:
Identified a potential vulnerability, here’s a suggested change for The function `load_dataset` does not handle exceptions that may occur during the loading of the MNIST dataset, such as network issues or file corruption.:

```suggestion
def load_dataset():
    try:
        # load dataset
        (trainX, trainY), (testX, testY) = mnist.load_data()
    except Exception as e:
        raise RuntimeError('Failed to load MNIST dataset') from e
    # reshape dataset to have a single channel
    trainX = trainX.reshape((trainX.shape[0], 28, 28, 1))
    testX = testX.reshape((testX.shape[0], 28, 28, 1))
    # one hot encode target values
    trainY = to_categorical(trainY)
    testY = to_categorical(testY)
    return trainX, trainY, testX, testY
```